src/main/java/com/zuzka/myshowcode/service/OrderService.java

```diff
@@ -0,0 +1,119 @@
+package com.zuzka.myshowcode.service;
+
+import com.zuzka.myshowcode.dto.ItemRequest;
+import com.zuzka.myshowcode.dto.OrderRequest;
+import com.zuzka.myshowcode.entity.Order;
+import com.zuzka.myshowcode.enums.OrderState;
+import com.zuzka.myshowcode.repository.OrderRepository;
+import com.zuzka.myshowcode.repository.ProductRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class OrderService {
+
+    private OrderRepository orderRepository;
+    private ProductRepository productRepository;
+    private ModelMapper modelMapper;
+
+    @Value("${order.expiration.seconds}")
+    private int orderExpirationTime;
+
+    public OrderService(OrderRepository orderRepository, ProductRepository productRepository, ModelMapper modelMapper) {
+        this.orderRepository = orderRepository;
+        this.productRepository = productRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    public List<ItemRequest> addNewOrder(OrderRequest order) {
+        List<ItemRequest> missingItems = getMissingItems(order);
+        if(missingItems.isEmpty()){
+            blockProductsInStock(order);
+            orderRepository.save(modelMapper.map(order, Order.class));
+            return new ArrayList<>();
+        } else {
+            return missingItems;
+        }
+    }
+
+    public List<Order> getAllOrders() {
+        return (List<Order>) orderRepository.findAll();
+    }
+
+    public Optional<Order> getOrderById(Long id) {
+        return orderRepository.findById(id);
+    }
+
+    public void payOrder(Long id) {
+        var order = orderRepository.findById(id).orElseThrow();
+        order.setState(OrderState.PAID);
+        orderRepository.save(order);
+    }
+
+    public void cancelOrder(Long id) {
+        var order = orderRepository.findById(id).orElseThrow();
+        order.setState(OrderState.CANCELLED);
+        orderRepository.save(order);
+        unblockProductsInStock(order);
+    }
+
+    public void checkForExpiredOrders() {
```

Comment:
Celá tahle metoda má jeden problém, a to je tahání všech objednávek, tedy zbytečná zátěž na databázi. Správně by se to mělo řešit už na úrovni db, tedy mít v Repository nějakou `@Query` s where klauzulí `createdDate > ?` a do této metody posílat `LocalDateTime.now().minusSeconds(orderExpirationTime)`. Ta `@Query` metoda může být rovnou `DELETE` operace, nebo pouze `SELECT` a pak celý list mazat v rámci Repository.